test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1887,9 +1889,23 @@ protected Settings nodeSettings(int nodeOrdinal) {
             builder.put(TelemetrySettings.TRACER_FEATURE_ENABLED_SETTING.getKey(), true);
             builder.put(TelemetrySettings.TRACER_ENABLED_SETTING.getKey(), true);
         }
+
+        // Randomly set a replication strategy for the node.
+        if (useRandomReplicationStrategy()) {
+            builder.put(CLUSTER_REPLICATION_TYPE_SETTING.getKey(), randomBoolean() ? ReplicationType.DOCUMENT : ReplicationType.SEGMENT);
```

Comment:
If a test fails, how will developer know which replication strategy was in use?  Should you do some logging?

Reply:
Yes you are right, we need some logging here. I will add this in next commit. Thanks @kkmr for pointing this out.

Reply:
Instead of using random, would it be beneficial to test both of those 2 cases to get better coverage?

Reply:
@ansjcy thanks for chiming in. I don't think it adds much value to run all test suites with both segrep and docrep, because not every test is concerned or affected by replication strategy. Instead running test twice would increase the gradle check workflow time without adding much value. 

For test cases where we feel it would be affected or concerned by replication strategy, we are running them both with segrep and docrep. example: https://github.com/opensearch-project/OpenSearch/pull/11773 , https://github.com/opensearch-project/OpenSearch/pull/12307

This PR specifically targets the tests which are not concerned or affected about replication strategy, but at same time we are not completely ignoring them by not using a specific replication strategy and increasing coverage.

Reply:
Thanks for the explaination! Makes sense to me.